README.md

```diff
@@ -232,9 +232,31 @@ For a more complete, downloadable example including distributed traces between m
 
 The source code package tree is scanned. For each source code file, `dave/dst` is used to build an AST of the source code in the file.
 
-The AST is checked for package level functions or methods that have a `//dd:span` comment attached to them. A function or method annotated with `//dd:span` must have a `context.Context` as its first parameter, or an `*http.Request` as any parameter in order for a span to be automatically inserted into the code. The context or request is required for trace information to be passed through function calls in a Go program. If this condition is met, the `//dd:span` comment is scanned for tags and code is inserted as the first lines of the function.
+The AST is checked for package level functions or methods that have a `//dd:span` comment attached to them.
 
-Orchestrion also supports automatic tracing of the following libraries:
+### dd:span magic comment
+
+A function or method annotated with `//dd:span` must have a `context.Context` as its first parameter, or an `*http.Request` as any parameter in order for a span to be automatically inserted into the code.
+
+The context or request is required for trace information to be passed through function calls in a Go program. If this condition is met, the `//dd:span` comment is scanned and code is inserted as the first lines of the function.
+
+It's possible to add key-value pairs separated by `:` or arguments (including their fields) present in the function or method signature. 
+
+```go
+//dd:span foo2:bar2 type:request name req.Method
+func MyFunc2(name string, req *http.Request) {
+	//dd:startinstrument
+	req = req.WithContext(instrument.Report(req.Context(), event.EventStart, "function-name", "MyFunc2", "foo2", "bar2", "type", "request", "name", name, "req.Method", req.Method))
+	defer instrument.Report(req.Context(), event.EventEnd, "function-name", "MyFunc2", "foo2", "bar2", "type", "request", "name", name, "req.Method", req.Method)
+	//dd:endinstrument%s
+}
+```
+
+Key-value pairs will be passed as strings. Arguments' values are expected to be string too.
```

Comment:
I think it's not clear why and what we're doing with the arguments, it's not documented above either.

Reply:
@ahmed-mez Would moving this before the example better? Also, I'll think a clear way to explain that internally we are casting values to string, and dropping them if they aren't castable: https://github.com/DataDog/orchestrion/blob/f32b6b896f3a834115bb3318352039f7828e85dd/instrument/instrument.go#L31

Thanks for your feedback!

Reply:
I think we can keep the sentence here (although a review from the docs team could help here).
I just noticed we do not document anywhere in the readme why would a user need to pass this metadata, what's the difference between passing a key:value tag or an argument name, and how the orchestrion is using this metadata.

I also understand that the purpose of the readme is to give quick start examples and the information in there shouldn't be too overwhelming, I'm open to leaving this suggestion for later when we write detailed documentation on the datadog docs website. Wdyt?

Reply:
I agree, although I added a first sentence to introduce a basic description of what `dd:span` comments do.